internal/bootstrap/agentbinary.go

```diff
@@ -64,8 +63,7 @@ func PopulateAgentBinary(ctx context.Context, dataDir string, storage AgentBinar
 
 	logger.Debugf("Adding agent binary: %v", agentTools.Version)
 
-	// If the hash already exists, we don't need to add it again.
-	if err := storage.Add(ctx, bytes.NewReader(data), metadata); err != nil && !errors.Is(err, objectstoreerrors.ErrHashAlreadyExists) {
```

Comment:
For people following along, this is safe to do here because we're already checking in the storage.Add method. We shouldn't be leaking the objectstore errors at this level.